### PR TITLE
EmuHawk - fixed playmovie bug when system == GG or SG (caused by this…

### PR DESCRIPTION
**Pull request by Asnivor**
…: #1277 )

This fixes an small issue with playing GG or SG movies from bizhawk <=2.3.0 in the master branch, after the change made in #1277

Should be able to close off ticket #1297

### CODE REVIEW

**Comment by Asnivor** [comment]
hmm, hold of on merging this - I think it needs more work

**Comment by Asnivor** [comment]
@Glitcholog

Can you test the dev build here: https://ci.appveyor.com/project/zeromus/bizhawk-udexo/build/0.0.0.2355/artifacts

And let us know whether it fixes the issue with loading the newer GG movie files (that have a GG system ID in the bk2 header)?

**Comment by Glitcholog** [comment]
I have done the test, I was able to load movie made in that Bizhawk Dev for some game gear and sega SG-1000 games I had around.  Here the three games for each system I used for the test. One of the movie was about 5 minute and didn't have desync

GG
Mortal Kombat
Mortal Kombat II
Sonic the Hedgehog

SG
Elevator Action
Champion Tennis
Champion Baseball

**Comment by Asnivor** [comment]
@vadosnaprimer all looks good now. You OK with the merge?

File: BizHawk.Client.Common/movie/MovieSession.cs [hunk 1]
```diff
@@ -475,6 +475,10 @@ public void QueueNewMovie(IMovie movie, bool record, IEmulator emulator)
 			if (!record) // The semantics of record is that we are starting a new movie, and even wiping a pre-existing movie with the same path, but non-record means we are loading an existing movie into playback mode
 			{
 				movie.Load(false);
+
+                // handle GG and SG
+                if (movie.SystemID == "GG" || movie.SystemID == "SG")
+                    movie.SystemID = "SMS";
```
[the remark this answers is not in the document]
Comment (Asnivor) [reply BizHawk.Client.Common/movie/MovieSession.cs]: this ticket: #1246 was fixed in PR #1324.

However, the knock on effect of this fix was that GG and SG movies were no longer recognized when you tried to load them. 

This PR fixes that.

TLDR: the correct System ID is saved in the movie when you record, and Bizhawk (with this PR) now recognizes this system ID correctly when a movie is loaded..

[the remark this answers is not in the document]
Comment (vadosnaprimer) [reply BizHawk.Client.Common/movie/MovieSession.cs]: I'm just trying to understand which part of the infrastructure was initially wrong, what was then fixed, what was broken, and which of all that is fixed now. Please post the details about the infrastructure, so one doesn't have to read and compare all those issues and PRs.

[the remark this answers is not in the document]
Comment (Asnivor) [reply BizHawk.Client.Common/movie/MovieSession.cs]: Ok. I'll have that for you tomorrow :)

[the remark this answers is not in the document]
Comment (Asnivor) [reply BizHawk.Client.Common/movie/MovieSession.cs]: Right.

Originally, when you recorded a __GG__ movie, the embedded __header.txt__ file would be written with Platform as __SMS__:
image

When this movie was subsequently loaded back into bizhawk, the header information showed the platform as SMS highlighted in red (issue #1246). This I think is because the associated ROM was not recognised as SMS (as it was GG).

To resolve this (in PR #1277 ) I added another switch case for GG within the Global.SystemInfo method (as it was not present before):

https://github.com/TASVideos/BizHawk/blob/f436e2542630c1f0aea16cca4953b1aafeceb3b0/BizHawk.Client.Common/Global.cs#L90

I also modified MovieConversionExtensions.PopulateWithDefaultHeaderValues() so that the movie.SystemID was updated in the header.txt file to reflect the 'new' SystemID:

https://github.com/TASVideos/BizHawk/commit/f436e2542630c1f0aea16cca4953b1aafeceb3b0?diff=unified#diff-677bd49b667e14224c53d8865ea3d34dL322

image

However, this lead to issue #1297 . The original PR had only taken into account __saving__ movies, not loading them back in. So after the original PR you could only load a movie with a SystemID of GG or SG, if you __first__ edited the header file and changed the platform back to SMS.

This latest PR rectifies that problem by updating MovieSession.QueueNewMovie() in order to handle GG and SG.

In retrospect, this may not be the optimum way of doing things (I don't know what would be), but it fixes the current issue caused by the previous fix that I made for the previous issue.

:/

**Comment by vadosnaprimer** [comment]
Apparently this set of PRs is fixing the wrong thing the wrong way.

**Comment by YoshiRulz** [comment]
If the [buggy code was reverted](https://github.com/TASVideos/BizHawk/issues/1297#issuecomment-423661666), can this branch be deleted?
